test/lib/ansible_test/_internal/commands/sanity/validate_modules.py

```diff
@@ -159,7 +159,11 @@ def test(self, args: SanityConfig, targets: SanityTargets, python: PythonConfig)
                 temp_dir = process_scoped_temporary_directory(args)
 
                 with tarfile.open(path) as file:
-                    file.extractall(temp_dir)
+                    if hasattr(tarfile, 'data_filter'):
+                        # Remove this check when Python 3.11 is EOL
```

Comment:
Use a deprecation comment here (implemented by our own `deprecated` plugin for `pylint`):

```suggestion
                        # deprecated: description='extractall fallback without filter' python_version='3.11'
```